kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/AbstractFilesArtifactResolver.java

```diff
@@ -0,0 +1,191 @@
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+public abstract class AbstractFilesArtifactResolver extends ArtifactResolver {
+
+    private static final Logger log = LoggerFactory.getLogger(AbstractFilesArtifactResolver.class);
+
+    private ClassLoader classLoader;
+    private List<URL> jarRepository;
+    private List<URL> effectivePoms;
+    private PomParser pomParser;
+
+
+    AbstractFilesArtifactResolver(ClassLoader classLoader, AFReleaseId releaseId) {
+        this.classLoader = classLoader;
+        this.jarRepository = new ArrayList<>();
+        this.effectivePoms = new ArrayList<>();
+        init(releaseId);
+    }
+
+    public boolean isLoaded() {
+        return pomParser != null;
+    }
+
+    // initialize in jar repository
+    private void init(AFReleaseId releaseId) {
+        jarRepository = buildResources(name -> isInJarStructuredFolder(name, "jar"));
+        effectivePoms = buildResources(name -> isInJarStructuredFolder(name, "pom"));
+        pomParser = buildPomParser(releaseId);
+    }
+
+    abstract protected List<URL> buildResources(Predicate<String> predicate);
+
+    abstract protected URL getURL(String path);
+
+    public ClassLoader getClassLoader() {
+        return classLoader;
+    }
+
+    public List<URL> getJarRepository() {
+        return jarRepository;
+    }
+
+    public PomParser getPomParser() {
+        return pomParser;
+    }
+
+    public List<URL> getEffectivePoms() {
+        return effectivePoms;
+    }
+
+    private boolean isInJarStructuredFolder(String name, String type) {
+        String[] paths = new String[]{"BOOT-INF/classes/KIE-INF/", "KIE-INF/lib/"};
+        for (String path : paths) {
+            // a bit of a hack "contains call"
+            if (name.contains(path) && name.endsWith("." + type)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private PomParser buildPomParser(AFReleaseId releaseId) {
+        List<URL> url = effectivePoms.stream().filter(e -> e.getFile().endsWith(toFile(releaseId, "pom"))).collect(toList());
+        if (url.isEmpty()) {
+            return null;
+        }
+        String path = url.get(0).toExternalForm();
+        URL pomFile = getURL(path);
+        if (pomFile == null) {
+            log.warn("Maven pom not found in path {}", path);
+            return null;
+        }
+        try (InputStream pomStream = pomFile.openStream()) {
+            DefaultModelReader reader = new DefaultModelReader();
+            Model model = reader.read(pomStream, Collections.emptyMap());
+            // dependencies were resolved already creating the effective pom during kjar creation
+            return new PomParser() {
+
+                @Override
+                public List<DependencyDescriptor> getPomDirectDependencies(DependencyFilter filter) {
+                    List<DependencyDescriptor> deps = new ArrayList<DependencyDescriptor>();
```

Comment:
Class not needed inside diamond
```suggestion
                    List<DependencyDescriptor> deps = new ArrayList<>();
```

Reply:
done